### PR TITLE
[SPARK-30928][ML] Remove unnecessary MiMa excludes

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
https://issues.apache.org/jira/browse/SPARK-30928
remove unnecessary MiMa excludes

### Why are the changes needed?

When auditing binary incompatible changes for 3.0, I found several MiMa excludes are not necessary, so remove these.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
run dev/mima to check

### CODE REVIEW

**Comment by huaxingao** [comment]
retest this please

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -316,12 +310,6 @@ object MimaExcludes {
     ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.ml.feature.OneHotEncoderEstimator"),
     ProblemFilters.exclude[MissingTypesProblem]("org.apache.spark.ml.feature.OneHotEncoder"),
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.transform"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.getInputCol"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.getOutputCol"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.inputCol"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.setInputCol"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.setOutputCol"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.ml.feature.OneHotEncoder.outputCol"),
```
Comment (viirya) [inline project/MimaExcludes.scala]: Is this because of SPARK-29565?

Comment (huaxingao) [reply project/MimaExcludes.scala]: Yes :)

**Comment by huaxingao** [comment]
cc @srowen @zhengruifeng

**Review by srowen — COMMENTED** [review]
Looks OK if they are no longer necessary. This goes into 3.0, right?

**Comment by huaxingao** [comment]
Yes. This goes into 3.0. @srowen

**Review by viirya — APPROVED** [review]
Looks good as this passed MiMa test.

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
LGTM.

**Comment by srowen** [comment]
Merged to master/3.0

**Comment by huaxingao** [comment]
Thanks! @srowen @viirya @zhengruifeng

**Pull request closed without merging**